plugins/modem-manager/fu-firehose-updater.c

```diff
@@ -497,17 +540,17 @@ fu_firehose_updater_send_program_file(FuFirehoseUpdater *self,
 	FuChunk *chk;
 
 	chunks = fu_chunk_array_new_from_bytes(program_file, 0, 0, payload_size);
-	/* last block needs to be padded to the next payload_size,
+	/* last block needs to be padded to the next sector_size,
 	 * so that we always send full sectors */
 	chk = g_ptr_array_index(chunks, chunks->len - 1);
-	if (fu_chunk_get_data_sz(chk) != payload_size) {
+	if ((fu_chunk_get_data_sz(chk) % sector_size) != 0) {
 		g_autoptr(GBytes) padded_bytes = NULL;
-		g_autofree guint8 *padded_block = g_malloc0(payload_size);
-		g_return_val_if_fail(padded_block != NULL, FALSE);
-		memcpy(padded_block, fu_chunk_get_data(chk), fu_chunk_get_data_sz(chk));
-		padded_bytes = g_bytes_new(padded_block, payload_size);
+		gsize padded_sz = sector_size * (fu_chunk_get_data_sz(chk) / sector_size + 1);
```

Comment:
LGTM correctly says:

    Multiplication result may overflow 'unsigned int' before it is converted to 'gsize'.

I think converting payload_size and sector_size to gsize in the function declaration is probably a good idea.

Reply:
Ignore the snap failure for now, it's not exactly awesome how snaps are built now.